<launch>
    <env name="ROSCONSOLE_FORMAT" value="[${severity}][${thread}][${node}/${function}:${line}]: ${message}"/>

    <!-- Joystick Device Argument -->
    <arg name="joy_dev" default="/dev/input/js1"/>

    <!-- Joy Node -->
    <node pkg="joy" type="joy_node" name="joy_drive" clear_params="true" output="screen" respawn="true" >
      <param name="joy_dev" value="$(arg joy_dev)" />
      <param name="dev" value="/dev/input/js1" />
      <param name="deadzone" value="0.12" />
      <!-- OUTPUT TOPICS -->
      <remap from="joy" to="joy" />
    </node>

    <!-- Teleoperation Node -->
    <node pkg="simple_drive" type="drive_teleop" name="drive_teleop" clear_params="true" output="screen" respawn="true">
      <param name="servo_pan_speed" type="int" value="5"/>
      <param name="servo_pan_max" type="int" value="160"/>
      <param name="servo_pan_min" type="int" value="0"/>
      <!-- INPUT TOPICS -->
      <remap from="joy" to="joy" />
      <!-- OUTPUT TOPICS -->
      <remap from="teleop/cmd_vel" to="teleop/cmd_vel" />
      <remap from="servo_pos" to="servo_pos" />
      <remap from="move_base/cancel" to="move_base/cancel" />
    </node>
</launch>
